rdrf/rdrf/testing/behaviour/features/steps.py

```diff
@@ -848,3 +848,49 @@ def return_to_patientlisting(step):
     world.browser.get(
         find("//a[text()='Patient List']").get_attribute("href")
     )
+
+
+def find_multiple(xp):
+    """
+    Helper function to find multiple items with the same xpath.
+    Initially created so that a list of elements could be iterated over,
+    and a particular attribute extracted for each item.
+    """
+    element_list = world.browser.find_elements_by_xpath(xp)
+    return element_list
+
+
+@step('I check the available survey options')
+def check_proms_lists(step):
+    """
+    Check that the lists of options for surveys
+    and communication methods are correct.
+    """
+    # First check if on the PROMS page;
+    assert world.browser.current_url.endswith("/clinicalproms"),\
+        "Not on PROMS request page!"
+    # Create empty objects;
+    request_type_list = []
+    comms_type_list = []
+    # Press the add button and check if fades in;
+    find("//a[contains(@class, 'btn btn-info')]").click()
```

Comment:
One thought: it might be worth naming the xpath expressions with what they're meant to find:

e.g. ADD_BUTTON = "//a[contains(@class, 'btn btn-info')]"

then find(ADD_BUTTON).click() tells us what you're trying to do

One trick I use sometimes is to emulate "enums" by grouping names as static members of a class like this:

class Colours:
    RED = 1
    BLUE = 2

so then you can say if house_colour == Colours.RED etc

In same way portions of the page could be grouped with named xpath expressions maybe